### PR TITLE
upgrade: node upgrade cleanup

### PR DESCRIPTION
**Pull request by MaximilianMeister**

### CODE REVIEW

**Comment by jsuchome** [comment]
Also, I like saving the error message on error, but don't we end with multiple attempts of ending the step (one from `save_error_state`, one from `nodes`) ?

**Comment by jsuchome** [comment]
[(8) Use sub-steps during the last part of the upgrade, part I: controller nodes](https://trello.com/c/StuV7nHt/203-8-use-sub-steps-during-the-last-part-of-the-upgrade-part-i-controller-nodes)

File: crowbar_framework/app/models/api/upgrade.rb [hunk 1]
```diff
@@ -557,8 +555,11 @@ def save_upgrade_state(message = "")
       end
 
       def save_error_state(message = "")
```
Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: What if we'd just raise an exception (with `message`) from `save_error_state`, catch it in `api/upgrade.rb` (nodes method) and call `end_step` once we catch the exception?

Than we could remove all those checks for `false` when calling methods from `api/node.rb` ...

**Comment by jsuchome** [comment]
[(8) Use sub-steps during the last part of the upgrade, part I: controller nodes](https://trello.com/c/StuV7nHt/203-8-use-sub-steps-during-the-last-part-of-the-upgrade-part-i-controller-nodes)

File: crowbar_framework/app/models/api/upgrade.rb (hunk 1 above)
Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: Something like https://github.com/crowbar/crowbar-core/pull/960

Comment (MaximilianMeister) [inline crowbar_framework/app/models/api/upgrade.rb]: i dropped the last commit now, so we can base #960 on top

**Comment by MaximilianMeister** [comment]
rebased

**Comment by MaximilianMeister** [comment]
@rhafer there was a commit that has been removed in the meantime, because it is now covered by one of @jsuchome PR's, I changed the title now

**Comment by MaximilianMeister** [comment]
rebased again

**Comment by MaximilianMeister** [comment]
rebased again

**Pull request merged**